src/main/java/ma/hibernate/dao/PhoneDaoImpl.java

```diff
@@ -12,11 +20,38 @@ public PhoneDaoImpl(SessionFactory sessionFactory) {
 
     @Override
     public Phone create(Phone phone) {
-        return null;
+        Session session = null;
+        Transaction transaction = null;
+        try {
+            session = factory.openSession();
+            transaction = session.beginTransaction();
+            session.persist(phone);
+            transaction.commit();
+            return phone;
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new DataProcessingException("Can't create phone", e);
```

Comment:
I think this is unnecessary to create a custom Exception in this task

Reply:
agree